geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySender.java

```diff
@@ -185,6 +187,19 @@
 
   protected volatile ConcurrentLinkedQueue<EntryEventImpl> tmpDroppedEvents =
       new ConcurrentLinkedQueue<>();
+
+  /**
+   * Contains wan replication events that were dropped by parallel gateway senders.
+   * Activate this hook by setting system property <code>ENABLE_TEST_HOOK_TEMP_DROPPED_EVENTS</code>
+   */
+  private volatile ConcurrentLinkedQueue<EntryEventImpl> testHookTempDroppedEvents;
+
+  /**
+   * Only used for testing purpose. This property enables test hook which collects all
+   * wan replication events that are dropped by parallel gateway senders.
+   */
+  private static final boolean ENABLE_TEST_HOOK_TEMP_DROPPED_EVENTS =
+      Boolean.getBoolean("enable-test-hook-temp-dropped-events");
```

Comment:
Please try to think of some ways to test dropped events without having to resort to a test hook. Test hooks in the product code should always be the very last resort. It would be better to refactor the class, wrap it with a decorator, introduce a factory that can wrap all instances with an optional decorator, etc.

In general, I recommend very detailed white-box testing at the unit test level. The system test level (includes dunit/distributed tests) should be lean more towards higher level black-block testing.